src/main/java/org/sonarlint/intellij/telemetry/TelemetryEngineProvider.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.telemetry;
+
+import com.intellij.openapi.application.PathManager;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.sonarlint.intellij.SonarApplication;
+import org.sonarsource.sonarlint.core.telemetry.Telemetry;
+
+public class TelemetryEngineProvider {
+  private static final String PRODUCT = "SonarLint IntelliJ";
+  private static final String STORAGE_FILENAME = "sonarlint_usage";
+
+  private final SonarApplication application;
+
+  public TelemetryEngineProvider(SonarApplication application) {
+    this.application = application;
+  }
+
+  public Telemetry get() throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00112)